src/main/java/com/nitnelave/CreeperHeal/block/CreeperChest.java

```diff
@@ -144,8 +147,11 @@ public void update()
                     i.setContents(both);
 
                 }
-                else
+                else {
+                    chest.setType(Material.CHEST, false); // False so it doesn't do any block physics
+                    chest.setData(data, false); // Set facing direction
                     ((InventoryHolder) chest.getState()).getInventory().setContents(storedInventory);
```

Comment:
You don't need to check for the second one, because if it's already set to chest (which happens before it continues on to the rest of the method, and if it failed to set it, it probably throw an error), and getState will automatically return a chest inventory holder object. For the first one, it's deprecated, but it works and the other way of it doing it is about 20 lines of code. I guess I can do it..?